sympy/vector/scalar.py

```diff
@@ -13,11 +13,17 @@ class BaseScalar(Symbol):
     """
 
     def __new__(cls, name, index, system, pretty_str, latex_str):
-        name = str(name)
-        pretty_str = str(pretty_str)
-        latex_str = str(latex_str)
         from sympy.vector.coordsysrect import CoordSysCartesian
-        obj = super(BaseScalar, cls).__new__(cls, name)
+        if isinstance(name, Symbol):
+            name = name.name
+        if isinstance(pretty_str, Symbol):
+            pretty_str = pretty_str.name
+        if isinstance(latex_str, Symbol):
+            latex_str = latex_str.name
+
+        index = _sympify(index)
+        system = _sympify(system)
+        obj = super(BaseScalar, cls).__new__(cls, Symbol(name), index, system, Symbol(pretty_str), Symbol(latex_str))
```

Comment:
I based this on @srjoglekar246's changes from https://github.com/sympy/sympy/pull/10130.